### ISSUE TITLE
Add support for Uint8Array's in place of node Buffer's to ipfs-unixfs-importer

### ISSUE THREAD

**Issue by Gozala** [ipfs/js-ipfs-unixfs#56]
I glanced through the implementation and it seems that dependence is inherited from use of following libraries

- [bl](https://www.npmjs.com/package/bl)
- https://github.com/multiformats/js-multihashing-async/issues/33

There are few lines that also use `Buffer` directly but there `Uint8Array` would have done just fine as well. 

Trying to actually remove here might be difficult, however we could probably just remove it from the API surface and turn `Uint8Array`s into `Buffer`s internally where needed.

**Comment by Gozala** [comment]
Actually I think I have overlooked chunk validation phase that seems to normalize each chunk

https://github.com/ipfs/js-ipfs-unixfs/blob/8e8d83d757276be7e1cb2581abd4b562cb8209e2/packages/ipfs-unixfs-importer/src/dag-builder/validate-chunks.js#L8-L19

Although I'm also starting to suspect that `ipfs.add([ new ArrayBuffer(1024) ])` would cause content was invalid error because `ArrayBuffer` because [`ArrayBuffer`s seem to be passed through](https://github.com/ipfs/js-ipfs/blob/62c1422dec35108cf3f20b7ae9460f984a7101c2/packages/ipfs-core-utils/src/files/normalise-input.js#L154-L231) although they do not have `.length` property.

**Comment by Gozala** [comment]
@achingbrain is this the right type signature for the importer:

```ts
declare function importer(source:AsyncIterable<ImportEntry>, ipld:IPLDResolver, options?:ImporterOptions)

type ImportEntry =
  | ImportFile
  | ImportDir

type ImportFile = {
  path: string|void,
  content: Content,
  mtime?: Mtime,
  mode?: Mode
}

type ImortDir = {
  path: string,
  content?:void,
  mtime?: Mtime,
  mode?: Mode
}


type Content =
  | Iterable<string>
  | Iterable<ArrayBuffer>
  | Iterable<ArrayBufferView>
  | AsyncIterable<string>
  | AsyncIterable<ArrayBuffer>
  | AsyncIterable<ArrayBufferView>

type Mode = string | number
type MTime = number | Date | UnixFSTime | HRTime
type UnixFSTime = {
  secs: number,
  nsecs?: number 
}
type HRTime = [number, number]

type IPLDResolver = {
 // ...
}

type ImporterOptions = {
  // ...
}
```

**Comment by achingbrain** [comment]
The type it expects to be yielded by the source is:

```
{
  path: String,
  mode: Number (optional),
  mtime: {
    secs: Number,
    nsecs: Number (optional)
  },  (optional)
  content: AsyncIterator<Buffer> (optional)
}
```

It handles more, as you found, maybe it shouldn't.

**Comment by Gozala** [comment]
In that case can we loosen up it a bit to accept `Uint8Array`s in place of buffers ?

**Comment by achingbrain** [comment]
Of course, we shouldn't require the input to be Buffers anywhere.  I think the only place that might get painful is the use of `bl`, which we have to avoid unnecessary copying, but as long as we end up with similar performance characteristics then all good.

**Comment by Gozala** [comment]
fixed by #69
